slasher/detection/attestations/spanner.go

```diff
@@ -149,13 +180,45 @@ func (s *SpanDetector) UpdateSpans(ctx context.Context, att *ethpb.IndexedAttest
 	// each validator in attesting indices.
 	for i := 0; i < len(att.AttestingIndices); i++ {
 		valIdx := att.AttestingIndices[i]
+		// Mark the double vote filter.
+		if err := s.markAttFilter(att.Data, valIdx); err != nil {
+			return errors.Wrap(err, "failed to update attestation filter")
+		}
 		// Update min and max spans.
 		s.updateMinSpan(source, target, valIdx)
 		s.updateMaxSpan(source, target, valIdx)
 	}
 	return nil
 }
 
+func (s *SpanDetector) markAttFilter(attData *ethpb.AttestationData, valIdx uint64) error {
```

Comment:
Can you please add a few comments in `markAttFilter`? 
I can imagine it's quite hard for an outsider to figure out what this function does

Reply:
Sure thing